### ISSUE TITLE
Phone Input Pattern SyntaxError

### ISSUE THREAD

**Issue by stefanbiba** [elementor/elementor#25389]
### Prerequisites

- [X] I have searched for similar issues in open and closed tickets and cannot find a duplicate.
- [X] I have troubleshooted my issue, and it still exists against the latest stable version of Elementor.

### Description

In Google Chrome Console an error is occurring when filling out a phone number in a phone input field:

`Pattern attribute value [0-9()#&+*-=.]+ is not a valid regular expression: Uncaught SyntaxError: Invalid regular expression: /[0-9()#&+*-=.]+/v: Invalid character in character class`

I guess the 'v' at the end should not be there :-)

### Steps to reproduce

Open a form, which was created with Elementor Pro, on any website with latest Elementor/Elementor Pro updates and fill the field for a tel number.

### Expected behavior

There should be no error in console.

### Isolating the problem

- [ ] This bug happens when only the Elementor (and Elementor Pro) plugins are active.
- [ ] This bug happens with the Hello Elementor theme active.
- [ ] I can reproduce this bug consistently by following the steps I described above.

### Elementor System Info

```txt
== Server Environment ==
	Operating System: Linux
	Software: nginx/1.24.0
	MySQL version: Ubuntu 22.04 v10.6.16
	PHP Version: 8.2.16
	PHP Memory Limit: 1024M
	PHP Max Input Vars: 10000
	PHP Max Post Size: 56M
	GD Installed: Yes
	ZIP Installed: Yes
	Write Permissions: All right
	Elementor Library: Connected

== WordPress Environment ==
	Version: 6.4.3
	Site URL: https://www.heurigendorf.at
	Home URL: https://www.heurigendorf.at
	WP Multisite: No
	Max Upload Size: 56 MB
	Memory limit: 768M
	Max Memory limit: 1024M
	Permalink Structure: /%postname%/
	Language: de-DE
	Timezone: Europe/Vienna
	Admin Email: wordpress@austria-design.at
	Debug Mode: Inactive

== Theme ==
	Name: Hello Elementor Child
	Version: 2.0.0
	Author: Elementor Team
	Child Theme: Yes
	Parent Theme Name: Hello Elementor
	Parent Theme Version: 3.0.1
	Parent Theme Author: Elementor Team

== User ==
	Role: administrator
	WP Profile lang: de_DE
	User Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/17.3.1 Safari/605.1.15

== Active Plugins ==
	ACF Content Analysis for Yoast SEO
		Version: 3.1
		Author: Thomas Kräftner, ViktorFroberg, marol87, pekz0r, angrycreative, Team Yoast

	Admin Columns
		Version: 4.6.1
		Author: AdminColumns.com

	Admin CSS MU
		Version: 2.10
		Author: Arun Basil Lal

	Advanced Custom Fields PRO
		Version: 6.2.7
		Author: WP Engine

	Advanced Post Queries
		Version: 1.1.1
		Author: Shabti Kaplan

	Borlabs Cookie - Cookie Opt-in
		Version: 2.2.67
		Author: Borlabs GmbH

	Classic Editor
		Version: 1.6.3
		Author: WordPress Contributors

	Custom links in Elementor Image Carousel
		Version: 1.1.1
		Author: Charlie Etienne

	Disable Comments
		Version: 2.4.5
		Author: WPDeveloper

	Dominant Color Images
		Version: 1.0.1
		Author: WordPress Performance Team

	Elementor
		Version: 3.19.4
		Author: Elementor.com

	Elementor Pro
		Version: 3.19.3
		Author: Elementor.com

	Imagify
		Version: 2.2
		Author: Imagify – Optimize Images & Convert WebP & Avif

	LoginPress
		Version: 3.0.5
		Author: LoginPress

	Make Elementor Default Editor
		Version: 0.1.4
		Author: Fletcher Digital

	Manage Notification E-mails
		Version: 1.8.6
		Author: Virgial Berveling

	Nginx Helper
		Version: 2.2.4
		Author: rtCamp

	Performance Lab
		Version: 2.9.0
		Author: WordPress Performance Team

	Performant Translations
		Version: 1.1.2
		Author: WordPress Performance Team

	Post SMTP
		Version: 2.8.11
		Author: Post SMTP

	Redis Object Cache
		Version: 2.5.1
		Author: Till Krüss

	Site Kit by Google
		Version: 1.121.0
		Author: Google

	Solid Security Basic
		Version: 9.3.1
		Author: SolidWP

	SVG Support
		Version: 2.5.5
		Author: Benbodhi

	WebP Uploads
		Version: 1.0.5
		Author: WordPress Performance Team

	White Label Branding for WordPress
		Version: 4.2.9.100641
		Author: Alberto Lau (RightHere LLC)

	WPFront User Role Editor Personal Pro
		Version: 3.2.0.11043
		Author: Syam Mohan

	WP Media folder
		Version: 5.8.2
		Author: Joomunited

	WP Rocket
		Version: 3.15.9
		Author: WP Media

	WP Rocket | Cache WP Rest API
		Version: 
		Author: WP Rocket Support Team

	WP Rocket | LazyLoad Threshold
		Version: 
		Author: WP Rocket Support Team

	Yoast Duplicate Post
		Version: 4.5
		Author: Enrico Battocchi & Team Yoast

	Yoast SEO
		Version: 22.2
		Author: Team Yoast


== Eigenschaften ==
	Custom Fonts: 1
	Custom Icons: 1

== Integrations ==
	
	google_maps: Aktiv
	recaptcha_v3: Aktiv


== Elementor-Experimente ==
	Verbessertes Laden von Assets: Aktiv
	Verbessertes Laden von CSS: Aktiv
	Inline-Schriftarten-Symbole: Aktiv
	Additional Custom Breakpoints: Inaktiv
	admin_menu_rearrangement: Standardmäßig deaktiviert
	Flexbox Container: Aktiv
	Upgrade Swiper Library: Aktiv
	Grid Container: Aktiv
	Default to New Theme Builder: Inaktiv
	Hello Theme Header & Footer: Inaktiv
	Editor Top Bar: Inaktiv
	Optimized Gutenberg Loading: Aktiv
	Build with AI: Inaktiv
	Startseiten: Inaktiv
	Nested Elements: Inaktiv
	Hintergrundbilder nachladen: Inaktiv
	Optimize Image Loading: Aktiv
	Globaler Styleguide: Inaktiv
	Page Transitions: Inaktiv
	Notes: Inaktiv
	Bedingungen anzeigen: Inaktiv
	Form Submissions: Inaktiv
	Scroll Snap: Inaktiv
	Menü: Inaktiv
	Taxonomy Filter: Aktiv


== Protokoll ==
	
PHP: showing 2 of 2PHP: 2024-03-06 15:22:39 [warning X 7][/var/www/vhosts/grossheuriger.at/httpdocs/heurigendorf.at/public/wp-content/plugins/elementor-pro/modules/loop-filter/query/taxonomy-manager.php::150] Undefined array key "logicalJoin" [array (
  'trace' => '
#0: /var/www/vhosts/grossheuriger.at/httpdocs/heurigendorf.at/public/wp-content/plugins/elementor-pro/modules/loop-filter/query/taxonomy-manager.php(150): Elementor\Core\Logger\Manager -> rest_error_handler()
#1: /var/www/vhosts/grossheuriger.at/httpdocs/heurigendorf.at/public/wp-content/plugins/elementor-pro/modules/loop-filter/query/taxonomy-query-builder.php(33): ElementorPro\Modules\LoopFilter\Query\Taxonomy_Manager -> get_hierarchy_of_selected_terms()
#2: /var/www/vhosts/grossheuriger.at/httpdocs/heurigendorf.at/public/wp-content/plugins/elementor-pro/modules/loop-filter/module.php(124): ElementorPro\Modules\LoopFilter\Query\Taxonomy_Query_Builder -> get_merged_queries()
#3: /var/www/vhosts/grossheuriger.at/httpdocs/heurigendorf.at/public/wp-includes/class-wp-hook.php(324): ElementorPro\Modules\LoopFilter\Module -> filter_loop_query()
#4: /var/www/vhosts/grossheuriger.at/httpdocs/heurigendorf.at/public/wp-includes/plugin.php(205): WP_Hook -> apply_filters()
',
)]
PHP: 2024-03-06 18:29:25 [warning X 2][/var/www/vhosts/grossheuriger.at/httpdocs/heurigendorf.at/public/wp-content/plugins/elementor-pro/modules/dynamic-tags/acf/tags/acf-color.php::34] Undefined array key 1 [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]

JS: showing 7 of 7JS: 2024-03-06 14:39:08 [error X 3][https://www.heurigendorf.at/wp-content/plugins/elementor/assets/lib/backbone/backbone.marionette.min.js?ver=2.4.5.e1:24:20082] View (cid: "view91") has already been destroyed and cannot be used. 
JS: 2024-03-06 14:39:09 [error X 4][https://www.heurigendorf.at/wp-content/plugins/elementor/assets/lib/backbone/backbone.marionette.min.js?ver=2.4.5.e1:24:20082] View (cid: "view140") has already been destroyed and cannot be used. 
JS: 2024-03-06 14:42:24 [error X 4][https://www.heurigendorf.at/wp-content/plugins/elementor/assets/lib/backbone/backbone.marionette.min.js?ver=2.4.5.e1:24:20082] View (cid: "view136") has already been destroyed and cannot be used. 
JS: 2024-03-06 14:44:52 [error X 1][https://www.heurigendorf.at/wp-content/plugins/elementor/assets/lib/backbone/backbone.marionette.min.js?ver=2.4.5.e1:24:20082] View (cid: "view4203") has already been destroyed and cannot be used. 
JS: 2024-03-06 14:50:36 [error X 2][https://www.heurigendorf.at/wp-content/plugins/elementor/assets/lib/backbone/backbone.marionette.min.js?ver=2.4.5.e1:24:20082] View (cid: "view124") has already been destroyed and cannot be used. 
JS: 2024-03-06 17:58:05 [error X 2][https://www.heurigendorf.at/wp-content/plugins/elementor/assets/lib/backbone/backbone.marionette.min.js?ver=2.4.5.e1:24:20082] View (cid: "view90") has already been destroyed and cannot be used. 
JS: 2024-03-06 18:39:27 [error X 1][https://www.heurigendorf.at/wp-content/plugins/elementor/assets/lib/backbone/backbone.marionette.min.js?ver=2.4.5.e1:24:20082] View (cid: "view120") has already been destroyed and cannot be used. 



== Elementor - Compatibility Tag ==
	
	Custom links in Elementor Image Carousel: Die Kompatibilität ist nicht angegeben
	Elementor Pro: Die Kompatibilität ist nicht angegeben
	Make Elementor Default Editor: Die Kompatibilität ist nicht angegeben

== Elementor Pro - Compatibility Tag ==
```

### Agreement

- [X] I agree that my issue may be closed without action if it doesn't meet all the requirements.

**Comment by nicholaszein** [comment]
Hello @stefanbiba! 

Thank you for submitting your issue! 🙏 

🏷️ We've labeled this issue and our team will take a look at it. If there is indeed a problem, they'll work on fixing it. We can't say exactly when it'll be sorted out, but we kindly ask for your patience. Thanks for understanding!

Best regards

**Comment by Ste-Hamill** [comment]
Can't use the form as a result of this?  I will use a different form instead.
Please sort a fix out ASAP

**Comment by Ste-Hamill** [comment]
It's related to the tel field.  If I change that to a text field the form works.

**Comment by Anukk** [comment]
I can confirm this error. But, the weird thing is that for my client, Chrome is the only browser were the form works. In Edge and Firefox he gets the regex error. Instead, for me it works in all browsers tested: Brave, Chrome, Firefox, Edge, Anyway, Ste-Hamill "fix" works! I changed the tel field to text and now the form works in my clients Firefox, too.

**Comment by caleb-bright** [comment]
I have come across this same issue. Hopefully, it will be fixed sometime.

**Comment by LeonardoPessatti** [comment]
Same problem. Hopefully, it will be fixed someday.

**Comment by kaketusse** [comment]
I have the exact same problem !
The regex used in the default input field for tel type is wrong...

Hopefully, it can be fixed...

**Comment by Cyrille-Gautier** [comment]
Same problem, it works if I replace pattern in JS with [0-9#&+*-=.]+
Hope it is fixed soon, Elementor Forms are such a great tool !!

**Comment by Darkace01** [comment]
Which part of the file did you relace please @Cyrille-Gautier 
I am experiencing the same issue

**Comment by Darkace01** [comment]
So what I discovered was causing the issue was the from email, confirm that the from email is on the same domain as the site or smtp ( if you use a custom smtp)

**Comment by pfigz** [comment]
I have multiple websites that use the form widget, but only one of them throws this console error. Tried changing the From email as described above by Darkace01 and still getting the regex error.

**Comment by agileware-justin** [comment]
Happy new year, 2025! Still a problem with Elementor today:
Elementor, Version 3.26.4
Elementor Pro, Version 3.26.3

**Comment by thenetireland** [comment]
LEAVE US A MESSAGE
Name: David Lawless
Email: email@emailaddress.ie
Tel: 123456789 (set to text field instead removes the Pattern Syntax error)

Your submission failed because of an error.

From email address is the same url as the site.

**Comment by bugnumber9** [comment]
Same issue.

**Comment by cicoph** [comment]
Please for Tel number add this pattern: `[0-9\(\)\#\&amp;\+\*\=\.\- ]+`
It will fix all browser support (and future update)

**Comment by gursuj** [comment]
Based on the [MDN page for the thrown error](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Errors/Regex_invalid_char_in_class), it's because some characters like `()-` need to be escaped in regex v-mode. 

For now, you can add an HTML widget with this code to ensure only phone number characters are entered. **Note**: Here, escaping backslashes are escaped too so that they are inserted in html:
```html
<script>
  document.addEventListener('DOMContentLoaded', function() {
    document.querySelectorAll('.elementor-field-type-tel input').forEach(x => {
      x.setAttribute('pattern', "[0-9\\(\\)#&+*\\-=.]+");
    });
  })
</script>
```
Thanks @cicoph for suggesting escaping the characters. Seems it's not necessary to escape all characters, only those that have special meaning in v-mode.

**Comment by aronhawkins-4** [comment]
Still getting this error in 2026.

**Comment by fildenis** [comment]
I confirm that this error message is still there

**Comment by caleb-bright** [comment]
Yeah, I was hoping this issue would be fixed sooner. Hopefully this year when v4 releases.

**Comment by ricardovigatti** [comment]
+1. Need a fix ASAP

**Comment by alexcape** [comment]
Hi - this is still happening - please fix.

**Comment by maso-hash** [comment]
I encountered this issue about a week ago (2026). The error only appears when the TranslatePress plugin is active.

In the default language (Hungarian), the form works perfectly. However, when I translate the form into any other language, the error occurs consistently.

Here are the details of my setup and tests:

I tested multiple versions, but the issue persists.
The phone number field is removed — only an email field is used.
There is only one email recipient (admin).
The error occurs regardless of other form fields (name, phone, etc.).
The issue specifically appears only when "Actions After Submit" is set to send a notification email.

So in summary:

Hungarian (default language): works fine
Any translated version: triggers the error
The problem is tied to the email notification action after submission

Has anyone experienced something similar or found a workaround?

**Comment by LeonardoPessatti** [comment]
> I encountered this issue about a week ago (2026). The error only appears when the TranslatePress plugin is active.
> 
> In the default language (Hungarian), the form works perfectly. However, when I translate the form into any other language, the error occurs consistently.
> 
> Here are the details of my setup and tests:
> 
> I tested multiple versions, but the issue persists. The phone number field is removed — only an email field is used. There is only one email recipient (admin). The error occurs regardless of other form fields (name, phone, etc.). The issue specifically appears only when "Actions After Submit" is set to send a notification email.
> 
> So in summary:
> 
> Hungarian (default language): works fine Any translated version: triggers the error The problem is tied to the email notification action after submission
> 
> Has anyone experienced something similar or found a workaround?

My workaround was giving up on Elementor and using another method to run the form.
Two years since the creation of this issue and the devs dont seem to care.

**Comment by aksellno** [comment]
Just noticed this issue myself in a form on a website I was setting up. I saw several submissions which were supposed to be sent, but got this message: Your submission failed because of a server error.

**Comment by balazsnemeth** [comment]
Just noticed this issue myself in a form on a website I was setting up.This is a pitty not being fixed since 2024

**Comment by maso-hash** [comment]
I solved the issue by replacing the Elementor Form widget with Gravity Forms.

After configuring Gravity Forms correctly, everything works as expected. The setup is straightforward and didn’t require any complex workaround.

In my case, this completely resolved the problem, so it seems to be related to the Elementor Form rather than the rest of the page setup. Hope this helps someone else facing the same issue.
